homeassistant/components/autarco/config_flow.py

```diff
@@ -55,3 +56,45 @@ async def async_step_user(
             errors=errors,
             data_schema=DATA_SCHEMA,
         )
+
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Handle re-authentication request from Autarco."""
+        self._reauth_email = entry_data[CONF_EMAIL]
+
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle re-authentication confirmation."""
+        errors = {}
+
+        if user_input is not None:
+            client = Autarco(
+                email=user_input[CONF_EMAIL],
+                password=user_input[CONF_PASSWORD],
+                session=async_get_clientsession(self.hass),
+            )
+            try:
+                await client.get_account()
+            except AutarcoAuthenticationError:
+                errors["base"] = "invalid_auth"
+            except AutarcoConnectionError:
+                errors["base"] = "cannot_connect"
+            else:
+                return self.async_update_reload_and_abort(
+                    self._get_reauth_entry(),
+                    data_updates=user_input,
+                )
+        return self.async_show_form(
+            step_id="reauth_confirm",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_EMAIL, default=self._reauth_email): str,
+                    vol.Required(CONF_PASSWORD): str,
+                }
+            ),
```

Comment:
Do we use the existing scheme constantly or a new one?

What I also wonder is whether you let users change only the email and password or only the password, also given that the unique entry is determined based on your email address.

Reply:
Typically, you do a new schema which only has the password. If you want to allow them to change the e-mail then you also need to update the unique id (and check there isn't one already with the other e-mail).

Reply:
Can't really make a good decision about that, it seems logical to me that you should always be able to change your email and password 🤔

Reply:
I typically think that if it's possible to change the e-mail address in the app/whatever other you can use with the service. Then we probably should allow it here too and handle it e.g. replace the unique id with a new e-mail address.
But then it also means the e-mail address is perhaps not such a good unique id and we should check it's actually the same account it's connecting too.

Reply:
The email address is the most unique thing you can use in this case, there is no special account ID etc. that the API provides. I also think it's fine if you can only change the password, the chance that someone actually needs to change their email address is small and otherwise you just have to add the integration to Home Assistant again.